library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/BlockingQueueHandler.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.async
+
+import androidx.annotation.MainThread
+import androidx.annotation.WorkerThread
+import com.datadog.android.sessionreplay.internal.processor.RecordedDataProcessor
+import com.datadog.android.sessionreplay.internal.processor.RumContextDataHandler
+import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
+import com.datadog.android.sessionreplay.internal.utils.TimeProvider
+import java.lang.ClassCastException
+import java.lang.NullPointerException
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.RejectedExecutionException
+
+internal class BlockingQueueHandler(
+    private val processor: RecordedDataProcessor,
+    private val rumContextDataHandler: RumContextDataHandler,
+    private val executorService: ExecutorService,
+    private val timeProvider: TimeProvider
+) {
+    // region internal
+
+    internal var workQueue = LinkedBlockingQueue<BlockingQueueItem>()
```

Comment:
does it need to be `var`? I don't see any other assignment. Probably can be just `val`.

```suggestion
    internal val workQueue = LinkedBlockingQueue<BlockingQueueItem>()
```